server/pbenchinacan/container-build.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+#
+# This script builds a container which, when run, starts a Pbench Server.
+#
+# Note:  successfully running this script required adding AUDIT_WRITE (for the
+#        operation of sshd) and CAP_SETFCAP (for the installation of httpd) to
+#        the list of default capabilities in /etc/containers/containers.conf
+#        (and, if that file doesn't exist, you'll need to create it with the
+#        other default capabilities, e.g., see
+#        https://man.archlinux.org/man/containers.conf.5.en#CONTAINERS_TABLE and
+#        https://github.com/containers/common/blob/da56e470c0c57c27e91bdc844b32c5dab6611394/pkg/config/containers.conf#L48)
+#
+
+set -o errexit
+
+# Locations inside the container
+INSTALL_ROOT=/opt/pbench-server
+SERVER_LIB=${INSTALL_ROOT}/lib
+SERVER_BIN=${INSTALL_ROOT}/bin
+CONF_PATH=${SERVER_LIB}/config/pbench-server.cfg
+HOSTNAME_F=pbenchinacan
+
+# Locations on the host
+GITTOP=${GITTOP:-$(git rev-parse --show-toplevel)}
+PBINC_DIR=${GITTOP}/server/pbenchinacan
+
+# Open a copy of the base container.  Docker format is required in order to set
+# the hostname.
+container=$(buildah from --format=docker quay.io/pbench/pbench-devel-rhel8:main)
+
+# We could mount the container filesystem and access it directly, but we
+# instead access it with buildah commands.
+# mnt=$(buildah mount $container)
+
+# Consider adding -v datavolume for the Server data, and perhaps SQL and ES data
+buildah config \
+    --label maintainer="Nick Dokos <ndokos@redhat.com>" \
+    --hostname $HOSTNAME_F \
+    --stop-signal SIGINT \
+    --port 22:55555  `# sshd` \
+    --port 8001      `# pbench-server` \
+    $container
+
+# Set up Pbench DNF repo and install the Server and Apache
+buildah copy $container ${PBINC_DIR}/etc/yum.repos.d/pbench.repo /etc/yum.repos.d/pbench.repo
+buildah run $container dnf install -y pbench-server httpd
+buildah run $container dnf clean all
+
+# Skip installing and configuring the Firewall
+
+# Work around a problem with cron running jobs as other users in a container
+# FIXME:  it's unclear whether abusing anything other than crond's file helps.
```

Comment:
Yeah....  There are a bunch of recurring warnings in the Journal log, and, while this kind of screwing around with PAM changes them, it doesn't actually make them go away.  So, while we do need to abuse the `cron` configuration here in order to allow it to run our jobs as `pbench`, we might want to back off from doing it to _all_ the PAM configurations.  I don't really know.

Reply:
Have you tried changing *just* the crond file? If that works (for some value of "works"), then limiting it to just that file seems preferable to me.

Reply:
I think Nick's reply amplifies that some additional investigation here would be good. Either in this PR or in a followup (recorded with an issue) would be fine by me, but for now I'm leaving it open.

Reply:
The problem is the definition of "works".  Without applying any abuse, things the Pbench Server Cron jobs fail to start; abusing just `crond` allows the Server's jobs to run, but there are various errors/warnings in the journal log; abusing the lot of them improves/changes the errors/warnings, but doesn't entirely remove them.

I believe that what I'm doing is reasonable, because I believe that the module whose referencdes I'm "removing" is missing/not-loaded.  But, I can certainly dial it back to just targeting `crond`.

Reply:
I thought I commented on this but cannot find the comment :-(

I don't know what `pam_loginuid` does (I read the manpage and I still don't know what it does in the large - I guess I need to read more on pam) and how much it matters, so I'm OK with it if you are OK with it, but I think at some point we need to find out more about it and see if we can quiet the messages. So Dave's suggestion of opening an issue and not worrying about it here, sounds good to me. Consider me "resolved" :-)

Reply:
I believe that `pam_loginuid` records the "original" UID associated with a process:  that is, when a process is created and associated with a user, that user's UID is set and logged, allowing a later audit to trace back the ownership of any subprocesses to it.  Thus, any facility which creates a process on behalf of another user (e.g., `login`, `sshd`, `crond`, et al.) should invoke it (but, `sudo` and `su` should _not_, because their actions should be associated with their invoking UID, not the set-to one).

I'm surmising that the PAM module and its effects are superfluous or otherwise undesirable inside a container under Podman, because the auditable stuff all happens _outside_ the container....

Nevertheless, I'll back off on the abuse to just `crond`, and I'll hold my nose when looking at the journal log.

Reply:
The remainder is now tracked in #2590.